### DISCUSSION TITLE
Responder node gets the IKE_DOWN EVENT without any corresponding IKE_UP EVENT

### DISCUSSION THREAD

**Discussion by hbhokare** [strongswan/strongswan#1096] [Q&A]
Hi,

We are trying to develop the IPSec based application using different strongswan/davici APIs. We are using CA signed certificate authentication and calling init tunnel api of Strongswan from one of the site based on business logic in application. We dont have CA authority present on responder hence we are getting "received AUTHENTICATION_FAILED" error on initiator site and "no trusted RSA public key found" error on responder site in charon log. On responder we are observing the IKE_DOWN event with unique id for which we never got IKE_UP event. Is it expected in case of responder who is never initiating the tunnel from strongswan side?  

Here are the event log and config details from responder site

Responder event details:
---------------------------
connid 94 time Tue Jun 14 23:29:37 2022

  ike-updown {
    A_B {
      uniqueid = 12
      version = 2
      state = CONNECTING
      local-host = 192.168.56.20
      local-port = 500
      local-id = %any
      remote-host = 192.168.56.19
      remote-port = 500
      remote-id = C=IN, ST=Maharashtra, L=Pune, O=ABC, OU=Systems, CN=SVC_CA_Signed_cluster01, E=support@abc.com
      initiator-spi = b5f1840eafb066b1
      responder-spi = 77cc7c1739684722
      encr-alg = AES_CBC
      encr-keysize = 128
      integ-alg = HMAC_SHA2_256_128
      prf-alg = PRF_AES128_XCBC
      dh-group = ECP_256
      tasks-passive [
        IKE_AUTH
        IKE_CERT_POST
        IKE_CONFIG
        CHILD_CREATE
        IKE_AUTH_LIFETIME
        IKE_MOBIKE
      ]
    }
  }

Responder charon log 
------------------------

Tue Jun 14 23:29:37 2022 11[CFG] added vici connection: A_B
Tue Jun 14 23:29:37 2022 11[CFG] installing 'A_B'
Tue Jun 14 23:29:37 2022 06[IKE] <12> 192.168.56.19 is initiating an IKE_SA
Tue Jun 14 23:29:37 2022 06[CFG] <12> selected proposal: IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_AES128_XCBC/ECP_256
Tue Jun 14 23:29:37 2022 12[IKE] <12> received 1 cert requests for an unknown ca
Tue Jun 14 23:29:37 2022 12[CFG] <12> looking for peer configs matching 192.168.56.20[%any]...192.168.56.19[C=IN, ST=Maharashtra, L=Pune, O=ABC, OU=Systems, CN=SVC_CA_Signed_cluster01, E=support@abc.com]
Tue Jun 14 23:29:37 2022 12[CFG] <A_B|12> selected peer config 'A_B'
Tue Jun 14 23:29:37 2022 12[IKE] <A_B|12> no trusted RSA public key found for 'C=IN, ST=Maharashtra, L=Pune, O=ABC, OU=Systems, CN=SVC_CA_Signed_cluster01, E=support@abc.com'

Thanks,
-Harish

**Comment by tobiasbrunner** [comment]
I suppose it's possible that this happens in some error scenarios, i.e. a _down_ event is triggered just to be safe that one is triggered in case it was necessary (better one too many than missing one). It could probably be avoided if we added additional state checks before triggering the event (would probably affect several locations). Another option would be to make note of whether an _up_ event was ever triggered and then avoid _down_ events for SAs where that wasn't the case. But not sure if it's worth or even necessary to do anything about that as a superfluous _down_ event shouldn't be a problem in general.
